skills/repo.go

```diff
@@ -127,8 +133,8 @@ func (r *Repository) AddSkill(skill *Skill) error {
 		return fmt.Errorf("cannot add skill, skill type %q is not known", skill.Type)
 	}
 	// Reject skill that don't pass type-specific sanitization
-	if err := t.Sanitize(skill); err != nil {
-		return err
+	if err := t.SanitizeSkill(skill); err != nil {
+		return fmt.Errorf("cannot add skill, integrity check failure: %v", err)
```

Comment:
`"cannot add skill: %v"`

No need to say "integrity check failure".. the error itself should say what is wrong with the skill.